flink-core/src/main/java/org/apache/flink/api/connector/source/lib/SourceReaderFactory.java

```diff
@@ -27,9 +27,17 @@
 /**
  * A factory for creating source reader instances.
  *
- * @param <OUT> The type of the output elements.
+ * @param <T> The type of the output elements.
  */
-public interface SourceReaderFactory<OUT, SplitT extends SourceSplit> extends Serializable {
-    /** Instantiates a new {@code SourceReader} using the provided context. */
-    SourceReader<OUT, SplitT> newSourceReader(SourceReaderContext readerContext);
+public interface SourceReaderFactory<T, SplitT extends SourceSplit> extends Serializable {
```

Comment:
Should be moved to the same package as the `Source` interface imo.

Reply:
Indeed. Moved.